packages/protocol/contracts/dapis/Api3Adminship.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "./interfaces/IApi3Adminship.sol";
+
+/// @title Adminship contract controlled by the API3 DAO
+contract Api3Adminship is IApi3Adminship {
+  string internal constant ERROR_UNAUTHORIZED = "Unauthorized";
+  string internal constant ERROR_ZERO_ADDRESS = "Zero address";
+  string internal constant ERROR_EXPIRATION_NOT_EXTENDED = "Expiration not extended";
+
+  /// @dev Meta admin sets the admin statuses of addresses and has super
+  /// admin privileges
+  address public metaAdmin;
+  mapping(address => AdminStatus) public adminStatuses;
+
+  /// @dev Reverts if the caller is not the meta admin or has admin/superadmin priviliges
+  modifier onlyAdminOrMetaAdmin() {
+    require(adminStatuses[msg.sender] >= AdminStatus.Admin || msg.sender == metaAdmin, ERROR_UNAUTHORIZED);
+    _;
+  }
+
+  /// @param _metaAdmin Address that will be set as the meta admin
+  constructor(address _metaAdmin) {
+    require(_metaAdmin != address(0), ERROR_ZERO_ADDRESS);
```

Comment:
Do we have to check for this? Since `_metaAdmin` is only provided by the user who deploys the contract, and they likely won't be providing the zero address.

Reply:
Also copied that over from API3Authorizer [contract](https://github.com/api3dao/airnode/blob/871f17741045d4ff6c47dfb4f81f897e6beaf1f2/packages/protocol/contracts/authorizers/Api3Authorizer.sol#L23) but I can't really make a strong case why it is needed other than being a good practice to check for "null" on arguments.

Reply:
I think it's unnecessary, but I'll let @bbenligiray share his take.

Reply:
It's commonly considered as being good practice